src/test/url.test.ts

```diff
@@ -0,0 +1,204 @@
+import { TestFactory } from './factory';
+import { HTTP_ERRORS } from '../utils';
+
+describe('Shortener', () => {
+  describe('GET /urls', () => {
+    const factory: TestFactory = new TestFactory();
+
+    beforeEach((done) => {
+      factory.init().then(done);
+    });
+
+    afterEach((done) => {
+      factory.close().then(done);
+    });
+
+    it('should return an empty array of urls', async () => {
+      const response = await factory.app.get('/urls');
+      expect(response.status).toBe(HTTP_ERRORS.OK);
+      expect(response.body).toEqual([]);
+    });
+
+    it('should return an array of urls', async () => {
+      const res1 = await factory.app.post('/shorten').send({ url: 'https://www.google.com' });
+      expect(res1.status).toBe(HTTP_ERRORS.CREATED);
+
+      const response = await factory.app.get('/urls');
+      expect(response.status).toBe(HTTP_ERRORS.OK);
+      expect(response.body.length).toBe(1);
+      expect(response.body[0].originalUrl).toBe('https://www.google.com');
+    });
+  });
+
+  describe('POST /urls', () => {
+    const factory: TestFactory = new TestFactory();
+
+    beforeEach((done) => {
+      factory.init().then(done);
+    });
+
+    afterEach((done) => {
+      factory.close().then(done);
+    });
+
+    it('should save url and return a new shortened url', async () => {
+      const response = await factory.app
+        .post('/shorten')
+        .send({ url: 'https://www.mail.google.com/mail/u/0/#inbox/FMfcgzGxTNvvQTsMtWlbcjkNvlmPkjnH' });
+
+      expect(response.status).toBe(HTTP_ERRORS.CREATED);
+      expect(response.body.message).toBe('Url shortened successfully');
+      expect(response.body.data).toHaveProperty('shortUrl');
+    });
+
+    it('should return an error if the request body is invalid', async () => {
+      const res1 = await factory.app.post('/shorten').send({});
+      expect(res1.status).toBe(HTTP_ERRORS.BAD_REQUEST);
```

Comment:
```suggestion
      expect(res1.status).toBe(HTTP_ERRORS.BAD_REQUEST);
      expect(res1.body.errors).toHaveLength(2);

      const res2 = await factory.app.post('/shorten').send({ url: 'google beats everything' });
      expect(res2.status).toBe(HTTP_ERRORS.BAD_REQUEST);
      expect(res2.body.errors).toHaveLength(1);
```

Reply:
Done. I didn't add the error messages to this